packages/expo/src/migrations/update-15-0-3/change-jest-preset.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  formatFiles,
+  logger,
+  ProjectConfiguration,
+  readProjectConfiguration,
+  stripIndents,
+  Tree,
+  updateJson,
+} from '@nrwl/devkit';
+import {
+  addPropertyToJestConfig,
+  removePropertyFromJestConfig,
+} from '@nrwl/jest';
+import { JestExecutorOptions } from '@nrwl/jest/src/executors/jest/schema';
+import { forEachExecutorOptions } from '@nrwl/workspace/src/utilities/executor-options-utils';
+import { join } from 'path';
+
+/**
+ * Change the preset in expo's jest config
+ * - remove transform and testRunner from jest config
+ * - changes preset from jest-expo to react-native
+ * - change to babel.config.json
+ */
+export default async function update(tree: Tree) {
+  let useJestExpo = false;
+  forEachExecutorOptions<JestExecutorOptions>(
+    tree,
+    '@nrwl/jest:jest',
+    (options, projectName) => {
+      if (options.jestConfig && tree.exists(options.jestConfig)) {
+        const jestConfig = tree.read(options.jestConfig, 'utf-8');
+
+        if (jestConfig.includes(`preset: 'jest-expo'`)) {
+          const project = readProjectConfiguration(tree, projectName);
+          changePreset(tree, project);
+          renameBabelJson(tree, project);
+          useJestExpo = true;
+        }
+      }
+    }
+  );
+
+  if (useJestExpo) {
+    removeJestExpoFromPackageJson(tree);
+  }
+
+  await formatFiles(tree);
+}
+
+function changePreset(host: Tree, project: ProjectConfiguration) {
+  const jestConfigPath = project.targets?.test?.options?.jestConfig;
+  if (!jestConfigPath || !host.exists(jestConfigPath)) return;
+  try {
+    removePropertyFromJestConfig(host, jestConfigPath, 'transform');
+    removePropertyFromJestConfig(host, jestConfigPath, 'testRunner');
+    removePropertyFromJestConfig(host, jestConfigPath, 'preset');
+    addPropertyToJestConfig(host, jestConfigPath, 'preset', 'react-native');
+  } catch {
+    logger.error(
+      stripIndents`Unable to update ${jestConfigPath} for project ${project.root}.`
+    );
+  }
+}
+
+function renameBabelJson(host: Tree, project: ProjectConfiguration) {
```

Comment:
Why do we need to rename `.babelrc` to `babel.config.json`?

Reply:
the reason being that we use react-native preset https://github.com/facebook/react-native/blob/main/jest-preset.js, which uses babel-jest: `'^.+\\.(js|ts|tsx)$': 'babel-jest',`
so it does not support babelrc be default, it could be done by ['babel-jest', , { configFile: '.babelrc'}], however, since it is not something i could change to, so i just change to babel.config.json

Reply:
got it, makes sense